pkg/kubelet/rkt/rkt.go

```diff
@@ -937,15 +985,21 @@ func (r *Runtime) preparePod(pod *api.Pod, pullSecrets []api.Secret) (string, *k
 		newUnitOption("Service", "ExecStopPost", markPodFinished),
 		// This enables graceful stop.
 		newUnitOption("Service", "KillMode", "mixed"),
+		// Track pod info for garbage collection
+		newUnitOption(unitKubernetesSection, unitPodUID, string(pod.UID)),
```

Comment:
@dcbw I think we already store these information in the annotations in the rkt pod manifest. I don't like we store information in multiple places. 
But this is good for now, I will need to do a rebase and clean up for #24647 after this gets in. cc @euank